inc/Modules/Plugin/Settings.php

```diff
@@ -0,0 +1,173 @@
+				'show_in_rest'      => [
+					'schema' => [
+						'type'       => 'object',
+						'properties' => [
+							'accessKey'  => [
+								'type' => 'string',
+							],
+							'secretKey'  => [
+								'type' => 'string',
+							],
+							'bucketName' => [
+								'type' => 'string',
+							],
+							'region'     => [
+								'type' => 'string',
+							],
+							'endpoint'   => [
+								'type'   => 'string',
+								'format' => 'uri',
+							],
+						],
+					],
+				],
+			],
+			self::OPTION_GITHUB_TOKEN   => [
+				'type'              => 'string',
+				'label'             => __( 'GitHub Token', 'oneupdate' ),
+				'description'       => __( 'Personal Access Token used to connect to GitHub.', 'oneupdate' ),
+				'sanitize_callback' => static function ( $value ): string {
+					return sanitize_text_field( $value );
+				},
+				'show_in_rest'      => true,
+			],
+		];
+
+		foreach ( $governing_settings as $key => $args ) {
+			register_setting(
+				self::SETTING_GROUP,
+				$key,
+				$args
+			);
+		}
+	}
+
+	/**
+	 * Get S3 credentials.
+	 *
+	 * @return array<string,string>
+	 */
+	public static function get_s3_credentials(): array {
+		$credentials = get_option( self::OPTION_S3_CREDENTIALS, [] );
+		if ( empty( $credentials ) || ! is_array( $credentials ) ) {
+			return [];
+		}
+		return [
+			'accessKey'  => isset( $credentials['accessKey'] ) ? Encryptor::decrypt( $credentials['accessKey'] ) : '',
+			'secretKey'  => isset( $credentials['secretKey'] ) ? Encryptor::decrypt( $credentials['secretKey'] ) : '',
+			'bucketName' => isset( $credentials['bucketName'] ) ? $credentials['bucketName'] : '',
+			'region'     => isset( $credentials['region'] ) ? $credentials['region'] : '',
+			'endpoint'   => isset( $credentials['endpoint'] ) ? $credentials['endpoint'] : '',
+		];
+	}
+
+	/**
+	 * Set S3 credentials.
+	 *
+	 * @param array<string,string> $credentials Credentials.
+	 * @return bool
+	 */
+	public static function set_s3_credentials( array $credentials ): bool {
+		$sanitized_credentials = [
+			'accessKey'  => isset( $credentials['accessKey'] ) ? Encryptor::encrypt( sanitize_text_field( $credentials['accessKey'] ) ) : '',
+			'secretKey'  => isset( $credentials['secretKey'] ) ? Encryptor::encrypt( sanitize_text_field( $credentials['secretKey'] ) ) : '',
+			'bucketName' => isset( $credentials['bucketName'] ) ? sanitize_text_field( $credentials['bucketName'] ) : '',
+			'region'     => isset( $credentials['region'] ) ? sanitize_text_field( $credentials['region'] ) : '',
+			'endpoint'   => isset( $credentials['endpoint'] ) ? esc_url_raw( $credentials['endpoint'] ) : '',
+		];
+		return update_option( self::OPTION_S3_CREDENTIALS, $sanitized_credentials );
+	}
+
+	/**
+	 * Get GitHub token.
+	 *
+	 * @return string
+	 */
+	public static function get_github_token(): string {
+		$token = get_option( self::OPTION_GITHUB_TOKEN, '' );
+		if ( empty( $token ) ) {
+			return '';
+		}
+		return Encryptor::decrypt( $token );
```

Comment:
::decrypt can return false.

```suggestion
		return false !== Encryptor::decrypt( $token ) ? $token : '';
```